substrate/substrate-data-raw/src/datasource.ts

```diff
@@ -42,20 +42,50 @@ export class RpcDataSource {
     public readonly rpc: Rpc
     private headPollInterval: number
     private newHeadTimeout: number
+    private finalityConfirmation?: number
     private log?: Logger
 
     constructor(options: RpcDataSourceOptions) {
         this.rpc = new Rpc(options.rpc)
         this.headPollInterval = options.headPollInterval ?? 5000
         this.newHeadTimeout = options.newHeadTimeout ?? 0
+        this.finalityConfirmation = options.finalityConfirmation
         this.log = options.log
     }
 
     async getFinalizedHeight(): Promise<number> {
-        let head = await this.rpc.getFinalizedHead()
-        let header = await this.rpc.getBlockHeader(head)
-        assert(header, 'finalized blocks must be always available')
-        return qty2Int(header.number)
+        if (this.finalityConfirmation == null) {
+            let head = await this.rpc.getFinalizedHead()
+            let header = await this.rpc.getBlockHeader(head)
+            assert(header, 'finalized blocks must be always available')
+            return qty2Int(header.number)
+        } else {
+            let retries = 0
+            while (retries < 5) {
+                let head = await this.rpc.getHead()
+                let header = await this.rpc.getBlockHeader(head)
```

Comment:
`chain_getHeader` can work without hash, so there is probably no need to call the two methods sequentially

Reply:
yes, that's right. thanks for the suggestion!